tests/foreman/ui/test_contenthost.py

```diff
@@ -1250,3 +1260,59 @@ def test_syspurpose_mismatched(session, vm_module_streams):
             'details'
         ]
         assert details['system_purpose_status'] == "Mismatched"
+
+
+@tier4
+def test_pagination_multiple_hosts_multiple_pages(session, module_host_template):
+    """Create hosts to fill more than one page, sort on OS, check pagination.
+
+    Search for hosts based on operating system and assert that more than one page
+    is reported to exist and that more than one page can be accessed. Make some
+    additonal aserts to ensure the pagination widget is working as expected.
+
+    To avoid requiring more than 20 fakes hosts to overcome default page setting of 20,
+    this test will set a new per_page default of 4, using URL method rather than the
+    "entries_per_page" setting to avoid impacting other tests.
+
+    :id: e63e4872-5fcf-4468-ab66-63ac4f4f5dac
+
+    :BZ: 1642549
+    """
+    new_per_page_setting = 3
+    host_num = new_per_page_setting + 1
+    host_name = None
+    start_url = f'/content_hosts?page=1&per_page={new_per_page_setting}'
+    # create more than one page of fake hosts. Need two digits in name to ensure sort order.
+    for count in range(host_num):
+        host_name = 'test-' + str('{:0>2}'.format(count))
```

Comment:
```suggestion
        host_name = 'test-' + str('{:0>2}'.format(count + 1))
```
This will give you `[1, 2, 3, 4]`, so you won't have to subtract 1 later on.

Also, string concatenation and `format()` look weird:
```python
        host_name = 'test-{:0>2}'.format(count + 1)
```

Or even use f-string:
```python
        host_name = f'test-{count + 1:0>2}'
```
(but with all these random numbers, I agree it's not the most readable code in the world)

Reply:
looks very tidy :) Will use that last example.